reference/cd-azure-devops.md

```diff
@@ -0,0 +1,277 @@
+---
+title: Azure DevOps
+---
+
+This page details how to use [Azure DevOps](https://azure.microsoft.com/en-us/services/devops/) to manage deploying
+stacks based on commits to specific Git branches, and based on the build reason. You may also choose to introduce a
+[Manual Intervention](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/manual-intervention?view=vsts) task to control the preview vs. update step for any pulumi stack.
+
+Pulumi doesn't require any particular arrangement of stacks or workflow to work in a
+continuous integration / continuous deployment system. So the steps described here can be
+altered to fit into any existing type of deployment setup.
+
+This page shows you how to use the YAML method to configure your DevOps pipeline, but you can easily adapt
+the steps outlined in the sample YAML file below to the Visual Designer as well.
+
+## Prerequisites
+- An account on https://app.pulumi.com.
+- The latest CLI.
+  - Installation instructions are [here](https://pulumi.io/quickstart/install.html).
+- A bare repo and set the remote URL to be your Azure DevOps project.
+  - To learn more about how to create a git repo in your DevOps project, click [here](https://docs.microsoft.com/en-us/azure/devops/organizations/projects/create-project?view=vsts&tabs=new-nav).
+
+## Stack and Branch Mappings
+The scripts below act on a hypothetical stack: `acme/acme-ui`.
+`acme-ui` contains the infrastructure code or `pulumi` program. It also contains an Angular-based SPA.
+The git repo for this look like this:
+```
+acme-ui
+  src/
+    app/
+      components/
+    index.html
+  infra/
+    index.ts
+    Pulumi.yaml
+    Pulumi.acme-ui.yaml
+  azure-pipelines.yml
+```
+Once you login into the `pulumi` CLI on your machine, you can create a stack by running `pulumi stack init`.
+To create a `pulumi` program using one of the many available templates, you may run `pulumi new <template>`.
+You can find a suitable template to run, using the command `pulumi new --help`.
+Learn more about the `pulumi` CLI commands [here](https://pulumi.io/reference/commands.html).
+
+Once your stack has been initialized, the `Pulumi.<stack-name>.yaml` file will be created with some basic configuration.
+The yaml file is used just for configuration values. All of your infrastructure will be built using your `pulumi` program.
+
+For this walkthrough, we will assume a `TypeScript`-based `pulumi` pr ogram, which will deploy resources to an Azure Subscription.
+
+**Note**: The names used above are purely for demonstration purposes only.
```

Comment:
Consider adding a `>` at the beginning of this line so that the code block gets formatted differently. Perhaps that would help differentiate this meta-point from the rest of the section?